packages/explat-client/src/create-explat-client.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Internal dependencies
+ */
+import { ExperimentAssignment, Config } from './types';
+import * as ExperimentAssignments from './internal/experiment-assignments';
+import * as Request from './internal/requests';
+import * as State from './internal/state';
+import * as Timing from './internal/timing';
+import * as Validation from './internal/validations';
+import { createNullExperimentAssignment } from './internal/experiment-assignments';
+
+/**
+ * The number of milliseconds before we abandon fetching an experiment
+ */
+const EXPERIMENT_FETCH_TIMEOUT = 2000;
```

Comment:
This is probably too short. In my testing the other day, I was seeing requests take 1.5 seconds occasionally. We don't want to accidentally bias our experiments with people that have fast/low-latency internet. With satellite internet, you can expect a 1kb file to take well over a second, sometimes 4 in bad weather. You can experience this with setting your network latency to about 900ms (which assumes a downlink somewhere decently far from the server) and throughput down to like 56kbps for a shared internet or maybe 128kbps if you want to emulate a rich family. Packet loss for a well positioned uplink is 5-10% and a storm adds another 10-15%.

This is probably the worst case, and I don't think we should optimize for it, but it might be better to let the network layer handle timeouts because it handles it very well and is well tested over all network conditions over the decades. We shouldn't be worried about it all, other than handling that case when it does occur.

Hopefully projects like starlink will help a ton. I do not miss satellite internet.

Reply:
Hmm, I've moved it up to 5 seconds, but I'm now thinking that maybe 2 seconds or less is better: 
- We don't timeout the actual request, it goes ahead and the result gets stored.
- If an experimenter is using it without advanced preloading a user could be waiting 5 seconds for their content to load and we don't want to trip Experimenters up on that.
- If an experimenter is using it with preloading 2 seconds should be plenty.

Reply:
> If an experimenter is using it without advanced preloading a user could be waiting 5 seconds for their content to load and we don't want to trip Experimenters up on that.

Chances are, if they're waiting more than five seconds for the experiment, they're waiting far more than 5 seconds for the whole page. My US phone is rate-limited to 256kbps because I'm in the EU (on an unlimited international plan) and there is nothing more annoying than application timeouts that don't consider this possibility when I have to use that phone off wifi. The platform endpoint takes more than 2 seconds in production, here's a snippet when connecting dev tools (calypso takes almost a minute fwiw):

![image](https://user-images.githubusercontent.com/1883296/106443405-bc220300-647c-11eb-8abf-89af382f55f2.png)

I'd like to see this removed, but five seconds is _probably_ ok if we're going to keep it.